Apps/Src/ReadCarCAN.c

```diff
@@ -20,43 +27,43 @@
 // CAN watchdog timer variable
 static OS_TMR canWatchTimer;
 
-// prechargeDlyTimer
+// Precharge delay timer variable
 static OS_TMR prechargeDlyTimer;
 
-// Array restart thread lock
-static OS_MUTEX arrayRestartMutex;
-static bool restartingArray = false;
-
 // NOTE: This should not be written to anywhere other than ReadCarCAN. If the need arises, a mutex to protect it must be added.
 // Indicates whether or not regenerative braking / charging is enabled.
-static bool chargeEnable = false;
+RCC_SCOPE bool chargeEnable = false;
 
 // Saturation buffer variables
 static int8_t chargeMsgBuffer[SAT_BUF_LENGTH];
-static int chargeMsgSaturation = 0;
+int chargeMsgSaturation = 0;
 static uint8_t oldestMsgIdx = 0;
 
-// SOC and Supp V
-static uint8_t SOC = 0;
-static uint32_t SBPV = 0;
-
+// Array ignition (IGN_1) pin status
+#ifdef __TEST_READCARCAN
+static bool arrayIgnitionStatus = true; // Can't test logic if ignition is always off
+#else
+static bool arrayIgnitionStatus = false;
+#endif
 
-// Handler to turn array back on
-static void arrayRestart(void *p_tmr, void *p_arg); 
+// SOC and Supp V
+uint8_t SOC = 0;
+uint32_t SBPV = 0;
 
+// Error assertion function prototype
+static void assertReadCarCANError(ReadCarCAN_error_code_t rcc_err);
 
 // Getter function for chargeEnable
-bool ChargeEnable_Get() 
-{
+bool ChargeEnable_Get(void){
```

Comment:
We can replace this with GETTER from common.h on master. Merge master into this branch to get the changes from there.